weave-js/src/components/Panel2/PanelPlot/PanelPlot.tsx

```diff
@@ -2797,11 +2796,33 @@ const PanelPlot2Inner: React.FC<PanelPlotProps> = props => {
         if (
           vegaReadyTable.columnSelectFunctions[dims.label].type !== 'invalid'
         ) {
-          newSpec.encoding.color.field = fixKeyForVega(dims.label);
+          newSpec.encoding.color.field = vegaCols[i].label;
           if (colorFieldIsRange) {
-            newSpec.encoding.color.scale = {
-              range: {field: fixKeyForVega(dims.color)},
+            // map the color field to the range of the color scale
+            const labelKey = vegaCols[i].label;
+            const colorKey = vegaCols[i].color;
+
+            const nonNullDims: ExprDimNameType[] =
+              PlotState.EXPRESSION_DIM_NAMES.filter(
+                dim =>
+                  !isVoidNode(
+                    series.table.columnSelectFunctions[series.dims[dim]]
+                  )
+              );
+
+            const mapping = flatPlotTable.reduce((acc, row) => {
+              if (nonNullDims.every(dim => row[vegaCols[i][dim]] != null)) {
+                acc[row[labelKey]] = row[colorKey];
+              }
+              return acc;
+            }, {} as {[key: string]: string});
```

Comment:
Not sure if this cast is needed - if so, I think the `Record<string, string>` notation is a bit more readable.